riscv/mmu.h

```diff
@@ -94,15 +94,19 @@ class mmu_t
           throw *matched_trigger;
       }
     }
-    if (unlikely(addr & (size-1))) {
-      if (require_alignment) load_reserved_address_misaligned(addr);
-      else return misaligned_load(addr, size, xlate_flags);
-    }
     if ((xlate_flags) == 0 && likely(tlb_load_tag[vpn % TLB_ENTRIES] == vpn)) {
+      if (unlikely(addr & (size-1))) {
+        if (require_alignment) load_reserved_address_misaligned(addr);
+        else return misaligned_load(addr, size, xlate_flags);
+      }
       if (proc) READ_MEM(addr, size);
       return from_target(*(target_endian<T>*)(tlb_data[vpn % TLB_ENTRIES].host_offset + addr));
     }
     if ((xlate_flags) == 0 && unlikely(tlb_load_tag[vpn % TLB_ENTRIES] == (vpn | TLB_CHECK_TRIGGERS))) {
+      if (unlikely(addr & (size-1))) {
+        if (require_alignment) load_reserved_address_misaligned(addr);
+        else return misaligned_load(addr, size, xlate_flags);
+      }
```

Comment:
This duplicated paragraph is really unfortunate, and likely to lead to subtle bugs where turning on a trigger in a page changes the behavior of misaligneds. (Because someone in the future updated one copy of this code and neglected the other.)

Reply:
The strategy I suggested on my `simplify-misaligned` branch eliminates this redundancy.

Reply:
Agreed, but I was trying to keep the code fast and readable, although I didn't benchmark.

In any case, Andrew's changes solve this nicely and I think we should make that change after this merges.